src/errorMiddleware.js

```diff
@@ -0,0 +1,22 @@
+module.exports = (hooks = {}) => async (error, req, res, next) => {
+  if (hooks.beforeSend) {
+    try {
+      await hooks.beforeSend(req, res)
+    } catch (beforeSendError) {
+      return next(beforeSendError)
+    }
+  }
+  if (res.headersSent) {
+    return next(error)
+  }
+  res.status(500).send({
```

Comment:
Shouldn't we send an error code too? Would it be an option to use our own `createError` here?